hpc/jvergeyn-vacc-strategies.py

```diff
@@ -0,0 +1,681 @@
+    ts = np.concatenate((np.linspace(start_t, start_t+spread, num=N), 
+                         constant_period
+                         )) 
+    
+    doses = np.concatenate((smooth_dose(daily_dose, 'up', start_dose=start_dose, N=N),
+                            np.repeat(np.array(daily_dose), len(constant_period))
+                            ))
+    return ts, doses
+
+def find_nearest_idx(array, value):
+    array = np.asarray(array)
+    idx = (np.abs(array - value)).argmin()
+    return idx
+
+def calc_N_vacc(N_vacc, age, prev_age, VE_states, dose, leftover, flag):
+    if (VE_states[:,[age]] > leftover) and flag:
+        dose = dose-N_vacc[:,[prev_age]]
+        N_vacc[:,[age]] = dose 
+        flag=False
+        if (VE_states[:,[age]]-leftover) <= dose:
+            N_vacc[:,[age]] = VE_states[:,[age]]-leftover 
+            flag=True
+    return N_vacc[:,[age]], flag
+
+## Vaccination functions
+
+NH_8_to_vacc = NH['85+']*initial_states_sept['VE'][8]
+NH_7_to_vacc = NH['75+']*initial_states_sept['VE'][7]
+NH_6_to_vacc = NH['65+']*initial_states_sept['VE'][6]
+
+def vacc_strategy(t, states, param, d, NH, order, elder):
+    """
+    time-dependent function for vaccination strategy
+    
+    states : dictionary
+        model states, VE = vaccine eligible states = S+R+E+I+A
+    d : dictionary
+        daily number of doses for that month
+        daily vaccinated persons on immunity date = daily dose on vaccination date / 2
+    NH : dictionary
+        proportion of residents in nursing homes per age group
+    
+    """
+    
+    leftover = 100
+    
+    VE_states = states['VE'].copy()
+    
+    if VE_states.ndim == 1:
+        VE_states = VE_states[np.newaxis]
+
+    N_vacc = np.zeros(VE_states.shape)
+
+    delay = pd.Timedelta('30D')
+    
+    t1 = pd.Timestamp('2021-01-01') + delay
+    t2 = pd.Timestamp('2021-02-01') + delay
+    t3 = pd.Timestamp('2021-03-01') + delay
+    t4 = pd.Timestamp('2021-05-01') + delay
+    daily_dose_jan = d['jan']/2
+    daily_dose_feb = d['feb']/2
+    daily_dose_ma_ap = d['mar-apr']/2
+    daily_dose_may_aug = d['may-aug']/2
+    
+    if t < t1:
+        N_vacc = N_vacc
+    
+    elif t1 <= t < t2: # January : nursing homes + part of care personnel
+        t_num = (t-t1)/pd.Timedelta('1D')
+        end_phase1 = NH_8_to_vacc/daily_dose_jan
+        end_phase2 = end_phase1 + NH_7_to_vacc/daily_dose_jan
+        end_phase3 = end_phase2 + NH_6_to_vacc/daily_dose_jan
+        end_t2 = (t2-t1)/pd.Timedelta('1D')
+        if t_num < end_phase1:
+            ts, doses = smooth_dose_ts(daily_dose_jan, 0, end_phase1, spread=2)
+            N_vacc[:,[8]] = doses[find_nearest_idx(ts, t_num)]
+        elif t_num < end_phase2:
+            ts, doses = smooth_dose_ts(daily_dose_jan, end_phase1, end_phase2, spread=1)
+            N_vacc[:,[7]] = doses[find_nearest_idx(ts, t_num)]
+        elif t_num < end_phase3:
+            ts, doses = smooth_dose_ts(daily_dose_jan, end_phase2, end_phase3, spread=0.3)
```

Comment:
Je smooth toch enkel het begin van elke fase he? Wat doet `spread` net, is dat voor EN na de startdatum dat je het uitspreid? Of enkle erna? Hoe zijn de waarden bepaald?